src/Sarif.FunctionalTests/v2/ConverterTestData/PREfast/Expected-050.xml.sarif

```diff
@@ -12,21 +12,21 @@
           "mimeType": "text/x-cpp"
         }
       },
-      "logicalLocations": {
-        "f": {
+      "logicalLocations": [
+        {
+          "fullyQualifiedName": "f",
           "decoratedName": "?f@@YAXH@Z"
         },
-        "f-0": {
-          "name": "f",
+        {
           "fullyQualifiedName": "f",
           "decoratedName": "?f@@YAXPAUS@@@Z"
         }
```

Comment:
>} [](start = 8, length = 1)

I wonder whether we need some sort of "readable name", like `f()` or `f(int, string)` to help understand the `decoratedName`. (Here we have two `f`s that differ by their decorated name due to different signatures. Or perhaps a tool that has the information can just use that readable name for `name` (and the `fQLN` would be `Namespace1::Namespace2::f(int, string)`

Reply:
Yes. Presumably, this collision exists simply because someone hasn't sufficiently qualified the 'name' property. These collisions can occur in C++ due to failure to express the presence of some qualifiers (such as 'const') on various components. Same problem: resolves to unique linkable thing, has a distinct decorated name, but a shared undecorated name.

---
In reply to: [243852449](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243852449) [](ancestors = 243852449)